schema-registry/serdes/src/main/java/com/hortonworks/registries/schemaregistry/serdes/avro/AbstractAvroSnapshotDeserializer.java

```diff
@@ -156,23 +169,60 @@ protected Object buildDeserializedObject(byte protocolId,
         String schemaName = schemaMetadata.getName();
         SchemaVersionKey writerSchemaVersionKey = new SchemaVersionKey(schemaName, writerSchemaVersion);
         LOG.debug("SchemaKey: [{}] for the received payload", writerSchemaVersionKey);
-        try {
-            Schema writerSchema = getSchema(writerSchemaVersionKey);
-            if (writerSchema == null) {
-                throw new SerDesException("No schema exists with metadata-key: " + schemaMetadata + " and writerSchemaVersion: " + writerSchemaVersion);
+        Schema writerSchema = getSchema(writerSchemaVersionKey);
+        if (writerSchema == null) {
+            throw new SerDesException("No schema exists with metadata-key: " + schemaMetadata + " and writerSchemaVersion: " + writerSchemaVersion);
+        }
+        Schema readerSchema = readerSchemaVersion != null ? getSchema(new SchemaVersionKey(schemaName, readerSchemaVersion)) : null;
+
+        deserializedObj = deserializePayloadForProtocol(protocolId, payloadInputStream, writerSchema, readerSchema);
+
+        if (includeSchemaAndVersion) {
```

Comment:
no, its not specific - actually fixes as per commit comment the problem for all the serdes, its more about handling avro schema with connect.